src/main/java/gregtech/client/ClientProxy.java

```diff
@@ -205,14 +201,22 @@ public static void addMaterialFormulaHandler(ItemTooltipEvent event) {
                     chemicalFormula = FluidTooltipUtil.getFluidTooltip(FluidStack.loadFluidStackFromNBT(compound.getCompoundTag(FluidHandlerItemStack.FLUID_NBT_KEY)));
                 }
             }
-
-        // Water buckets have a separate registry name from other buckets
-        } else if (itemStack.getItem().equals(Items.WATER_BUCKET)) {
+        } else if (itemStack.getItem().equals(Items.WATER_BUCKET)) { // Water buckets have a separate registry name from other buckets
             chemicalFormula = FluidTooltipUtil.getWaterTooltip();
+        } else if (itemStack.getItem() instanceof MetaOreDictItem) {
+            MetaOreDictItem oreDictItem = (MetaOreDictItem) itemStack.getItem();
+            Optional<String> oreDictName = OreDictUnifier.getOreDictionaryNames(itemStack).stream().findFirst();
+            if (oreDictName.isPresent() && oreDictItem.FORMULA_TO_OREDICTITEM.containsKey(oreDictName.get())) {
```

Comment:
is this `containsKey` call correct? `FORMULA_TO_OREDICTITEM` is keyed by the chemical formula, but here you are passing in the oredict name of the item as the key?